src/phy.c

```diff
@@ -338,3 +338,164 @@ static int handle_ed_scan(struct nl802154_state *state,
 
 COMMAND(get, ed_scan, "[<page> [<channels> [<duration>]]]",
     NL802154_CMD_ED_SCAN_REQ, 0, CIB_PHY, handle_ed_scan, NULL);
+
+
+static int print_beacon_notify_indication(struct nl_msg *msg, void *arg)
+{
+	struct genlmsghdr *gnlh;
+	struct nlattr *tb_msg[NL802154_ATTR_MAX + 1];
+	unsigned int *wpan_phy = arg;
+	int r;
+	int i;
+	int sdu_len;
+	size_t len;
+	const int BEACON_SDU_LEN_MAX = 127;   // FIXME magic number
+	uint8_t sdu[BEACON_SDU_LEN_MAX];
+
+	gnlh = nlmsg_data( nlmsg_hdr( msg ) );
+	if ( NULL ==  gnlh ) {
+	    fprintf( stderr, "gnlh was null\n" );
+	    goto protocol_error;
+	}
+
+	r = nla_parse( tb_msg, NL802154_ATTR_MAX, genlmsg_attrdata( gnlh, 0 ),
+	      genlmsg_attrlen( gnlh, 0 ), NULL );
+	if ( 0 != r ) {
+	    fprintf( stderr, "nla_parse\n" );
+	    goto protocol_error;
+	}
+
+	printf("beacon_indication:\n");
+
+	if (tb_msg[NL802154_ATTR_BEACON_SEQUENCE_NUMBER]) {
+		printf("BSN: %x\n", nla_get_u32(tb_msg[NL802154_ATTR_BEACON_SEQUENCE_NUMBER]));
+	}
+
+	if (tb_msg[NL802154_ATTR_PAN_DESCRIPTOR]) {
+		struct nlattr *tb_pan_desc[NL802154_ATTR_MAX + 1];
+
+		static struct nla_policy pan_desc_policy[NL802154_ATTR_MAX + 1] = {
```

Comment:
Are you referring to the nl802154_policy in the kernel?  I was following the method used in print_phy_handler() in src/info.c